src/resources/resources.tsx

```diff
@@ -38,14 +38,25 @@ class Resources extends React.PureComponent<ResourcesProps> {
   }
 }
 
-export const mapState = (state: StoreState, props: OwnProps) => ({
-  resources:
-    state && props.player
-      ? state.resources
-      : state && !props.player
-      ? state.enemyResources
-      : resourceReducerDefaultState
-});
+export const mapState = (state: StoreState, props: OwnProps) => {
+  if (props.player) {
+    return {
+      resources: state ? state.resources : resourceReducerDefaultState
+    };
+  }
+
+  if (!props.player)
+    return {
+      resources: state
+        ? {
+            energy: state.enemyCombatant.energy,
+            max_energy: state.enemyCombatant.max_energy,
+            health: state.enemyCombatant.health,
+            max_health: state.enemyCombatant.max_health
+          }
+        : resourceReducerDefaultState
+    };
+};
 
 const connector = connect(mapState);
```

Comment:
Oh is this a re-used mapStateToProps connected component :O Pretty interesting, not really seen this pattern before... Alternative to check out is the `useSelector` hook which would just need to take a function to pass the state you want from your store and wouldn't require the component to be `connected`... personally find it a bit easier to parse https://redux.js.org/recipes/usage-with-typescript#typing-the-useselector-hook It does require you to use functional components though.

Reply:
Good course on re-usable state using Hooks... really worth a watching when you've got time: https://egghead.io/courses/reusable-state-and-effects-with-react-hooks